### PR TITLE
Cusini/feature/curve check for efem cases

### PR DESCRIPTION
**Pull request by CusiniM**
This PR

- [x] modifies the `curve_check` helper to handle the case of an empty set name
- [x] adds a curve_check step for `Sneddon` efem integratedTests
- [x] creates new links to the newly organized efem input files
- [x] adds 2 Sneddon tests that only rely on a curve check step and employ an iterative solver:
   - [x] A case that uses static condensation implemented in GEOS + boomer AMG
   - [x] A case that employs MGR to perform the static condensation 

It is related to https://github.com/GEOS-DEV/GEOS/pull/2537

### CODE REVIEW

File: scripts/geos_ats_package/geos_ats/helpers/curve_check.py [hunk 1]
```diff
@@ -230,47 +239,65 @@ def compare_time_history_curves(fname, baseline, curve, tolerance, output, outpu
...
         except Exception as e:
             errors.append(str(e))
 
     # Reshape data if necessary so that they have a predictable number of dimensions
     for k in data.keys():
         for p, s in curve:
+            if s=='empty_setName':
+                key=f'{p}'
+            else:
+                key=f'{p}{s}'
```
Comment (TotoGaz) [inline scripts/geos_ats_package/geos_ats/helpers/curve_check.py:280]:
```suggestion
            key = f'{p}'
            if s != 'empty_setName':
                key += f'{s}'
```

File: scripts/geos_ats_package/geos_ats/helpers/curve_check.py [hunk 2]
```diff
@@ -230,47 +239,65 @@ def compare_time_history_curves(fname, baseline, curve, tolerance, output, outpu
...
                 data_sizes[p][s][k].append(1)
 
     # Check data diffs
     size_err = '{}_{} values have different sizes: target=({},{},{}) baseline=({},{},{})'
     for p, set_data in data_sizes.items():
         for s, set_sizes in set_data.items():
+            if s=='empty_setName':
+                key=f'{p}'
+            else:
+                key=f'{p}{s}'
```
Comment (TotoGaz) [inline scripts/geos_ats_package/geos_ats/helpers/curve_check.py:296]: same

File: scripts/geos_ats_package/geos_ats/helpers/curve_check.py [hunk 3]
```diff
@@ -306,14 +333,46 @@ def curve_check_parser():
...
     """
+
+   # Custom action class
+    class PairAction(argparse.Action):
+        def __call__(self, parser, namespace, values, option_string=None):
+            pairs = getattr(namespace, self.dest)
+            if len(values) == 1:
+                pairs.append((values[0], 'empty_setName'))
+            elif len(values) == 2:
+                pairs.append((values[0], values[1]))
```
Comment (TotoGaz) [inline scripts/geos_ats_package/geos_ats/helpers/curve_check.py:344]: Can you log/kill if there's more than 2 values?

File: scripts/geos_ats_package/geos_ats/helpers/curve_check.py [hunk 4]
```diff
@@ -306,14 +333,46 @@ def curve_check_parser():
     Returns:
         argparse.parser: The curve check parser
     """
+
+   # Custom action class
+    class PairAction(argparse.Action):
+        def __call__(self, parser, namespace, values, option_string=None):
+            pairs = getattr(namespace, self.dest)
+            if len(values) == 1:
+                pairs.append((values[0], 'empty_setName'))
```
Comment (TotoGaz) [inline scripts/geos_ats_package/geos_ats/helpers/curve_check.py:342]: Can you define a constant `empty_setName`?

Comment (CusiniM) [reply scripts/geos_ats_package/geos_ats/helpers/curve_check.py:338]: do you mean like a global variable in this file?

Comment (TotoGaz) [reply scripts/geos_ats_package/geos_ats/helpers/curve_check.py]: yes, something like this.

File: scripts/geos_ats_package/geos_ats/helpers/curve_check.py [hunk 5]
```diff
@@ -306,14 +331,49 @@ def curve_check_parser():
...
+                
+                if values:  # If there are more values
+                    if len(values) >= 2:
+                        script_info.extend(values[:2])  # Append a pair of values
+                        values = values[2:]  # Remove the pair of values
+                    else:
+                        script_info.append(values[0])  # Append the single value
+                        script_info.append(DEFAULT_SET_NAME)  # Add default set name
+                        values = []  # Clear the remaining value
+                scripts.append(script_info)
```
Comment (cssherman) [inline scripts/geos_ats_package/geos_ats/helpers/curve_check.py:365]: Given that we only expecting length 3 or 4 arguments, I'd recommend simplifying this parsing to something like:

```python

N = len(values)
if (N < 3) or (N > 4):
  raise Exception ('The -s option requires 3 or 4 inputs')
elif N == 3:
  values.append(DEFAULT_SET_NAME)
```

I expect that the while loop could lead to some annoying debugging

**Review by cssherman — APPROVED** [review]

**Pull request merged**